operational/azure/total_instance_hours/azure_usage_instance_hours_used.pt

```diff
@@ -0,0 +1,412 @@
+
+    var matching_nfu_data = _.find(nfu_ratios, function(ratio) {
+      return data.instance_type == ratio.instance_type
+    })
+
+    if (!(matching_nfu_data == undefined && matching_nfu_data == null)) {
+      data["instance_family"] = matching_nfu_data.instance_family
+      data["normalized_instance_hours"] = data.usage_amount * matching_nfu_data.nfu
+    }
+  })
+
+  //For each month, sum Instance Hours for each Instance Family
+  var month = _.pluck(_.uniq(usage_data, function(data) { return data.month }), "month")
+  var inst_family = _.pluck(_.uniq(usage_data, function(data) { return data.instance_family }), "instance_family")
+
+  _.each(month, function(mo) {
+    _.each(inst_family, function(fam) {
+      var total_inst_hrs = 0
+      _.each(usage_data, function(data) {
+        if (data.month == mo && data.instance_family == fam) {
+          if (data.normalized_instance_hours == null) {
+            data.normalized_instance_hours = 0
+          }
+          total_inst_hrs += data.normalized_instance_hours
+        }
+      })
+      temp_result.push({
+        "month": mo,
+        "instance_family": fam,
+        "total_instance_hours": total_inst_hrs
+      })
+    })
+  })
+
+  //Get highest 8 Instance Families for Instance Hours used
+  var inst_hr_totals_per_fam = []
+  _.each(inst_family, function (fam) {
+    var total_inst_hrs_12_months = 0
+    _.each(temp_result, function (data) {
+      if (data.instance_family == fam) {
+        total_inst_hrs_12_months += data.total_instance_hours
+      }
+    })
+    inst_hr_totals_per_fam.push({
+      "total_inst_hrs_12_months": total_inst_hrs_12_months,
+      "instance_family": fam
+    })
+  })
+
+  var top_8_inst_fams = _.last(_.pluck(_.sortBy(inst_hr_totals_per_fam, "total_inst_hrs_12_months"), "instance_family"), [8])
+
+  //If Instance Family is not in 8 highest Hours used, then put into "Other" category
+  _.each(month, function(mo) {
+    var total_inst_hrs_other = 0
+    _.each(temp_result, function(data) {
+      if (data.month == mo) {
+        var exists = _.find(top_8_inst_fams, function(inst_fam) { return inst_fam == data.instance_family })
+        if (exists == null) {
+          total_inst_hrs_other += data.total_instance_hours
+        } else {
+          result.push({
+            "month": data.month,
+            "instance_family": data.instance_family,
+            "total_instance_hours": data.total_instance_hours
+          })
+        }
+      }
+    })
+    result.push({
+      "month": mo,
+      "instance_family": "Other",
+      "total_instance_hours": total_inst_hrs_other
+    })
+  })
+  EOS
+end
+
+#CHART CREATION
+datasource "ds_chart_creation" do
+  run_script $js_chart_creation, $ds_instance_hours_per_fam, $param_region
+end
+
+script "js_chart_creation", type: "javascript" do
+  parameters "inst_hrs_data", "param_region"
+  result "report"
```

Comment:
Value should be "result" or normalize it with other policies.

"result" for naked scripts, "request" for scripts producing an API request for a datasource

Reply:
Resolved